tfx/proto/infra_validator.proto

```diff
@@ -119,6 +119,46 @@ message KubernetesConfig {
   //
   // Should be a positive integer. Default to 24 hours (86400).
   int32 active_deadline_seconds = 2;
+
+  // Optional.
+  // Additional Kubernetes pod overrides to apply to the serving pod
+  PodManifestOverrides serving_pod_manifest_overrides = 3;
+}
+
+message PodManifestOverrides {
+  // Optional.
+  // Kubernetes annotations to apply to the pod
+  map<string, string> annotations = 1;
+
+  // Optional.
+  // Kubernetes environment variables to apply to the pod
+  map<string, string> environment_variables = 2;
+
+  // Optional.
+  // Mapping from Kubernetes secret to environment variable to apply to the pod
+  repeated PodSecretToEnvironmentMapping pod_secret_to_environment_mapping = 3;
+}
+
+message PodSecretToEnvironmentMapping {
```

Comment:
This message seems a bit cumbersome. Sorry for your extra work, but let's mimic the Kubernetes EnvVar:

```proto
message PodOverrides {
  map<string, string> annotations = 1;
  repeated EnvVar env = 2;
}
message EnvVar {
  string name = 1;
  string value = 2;
  EnvVarSource valueFrom = 3;
}
message EnvVarSource {
  SecretKeySelector secretKeyRef = 4;
  reserved 1, 2, 3;
}
message SecretKeySelector {
  string name = 1;
  string key = 2;
}
```

Reference: [V1EnvVar](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1EnvVar.md)

Reply:
I think it's valid point and that your proposal is easier and more intuitive for users. Will fix.

Reply:
Fixed this.